test/test_python_dispatch.py

```diff
@@ -551,6 +551,25 @@ def foo_impl(x, y, z, w):
             getattr(torch.ops, self.test_ns).foo.default,
             getattr(torch.ops, self.test_ns).foo_functional.default, (x, y, z, w))
 
+    def test_register_fallthrough(self):
+        try:
+            my_lib = Library('aten', 'IMPL')
+            my_lib.impl("mm", fallthrough_kernel, "AutocastCPU")
+
+            a = torch.randn(2, 3, device='cpu', dtype=torch.float32)
+            b = torch.randn(3, 2, device='cpu', dtype=torch.float32)
+            with torch.autocast(device_type="cpu", dtype=torch.bfloat16):
+                # dtype for mm should be float32 since we registered a fallthrough
+                self.assertTrue(torch.mm(a, b).dtype == torch.float32)
```

Comment:
nit: does self.assertEqual work on dtypes? If so we should use that for more informative error messages